### PR TITLE
Free temp memory no longer needed in multibyte_split processing

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Updates the `multibyte_split` logic to free temporary memory once the chars and offsets have been resolved. This gives room to the remaining processing if more temp memory is required.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by davidwendt** [comment]
The change is best viewed with the "Hide whitespace" option. A main section of the source was converted to a IILE which mainly indents many of the lines into a new bracket `{ }` scope.
github-pr-hide-whitespace

File: cpp/src/io/text/multibyte_split.cu [hunk 1]
```diff
@@ -55,6 +55,10 @@
 #include <numeric>
 #include <optional>
 
+namespace cudf {
+namespace io {
+namespace text {
```
Comment (bdice) [inline cpp/src/io/text/multibyte_split.cu:60]: I like combining these, it tells me more about the scope of this file (specifically that we won't be putting things into `cudf::io` or `cudf`, just `cudf::io::text`.
```suggestion
namespace cudf::io::text {
```

**Review by bdice — APPROVED** [review]
Looks like most of the diff is code movement/indentation. It looks like the only relevant change is scoping, to allow freeing the temporaries. This looks fine.

**Comment by bdice** [comment]
> The change is best viewed with the "Hide whitespace" option.

I saw this comment just a bit too late! Thanks for the heads-up.

**Review by nvdbaranec — APPROVED** [review]
Only comment I have is that this function is fairly bulky. Might be cleaner to separate this code out into it's own function instead of embedded as a lambda.  Minor point though.

**Comment by davidwendt** [comment]
> Only comment I have is that this function is fairly bulky. Might be cleaner to separate this code out into it's own function instead of embedded as a lambda. Minor point though.

Yes, I usually prefer that approach myself but the number of parameters passed in and out seemed like too much here.

**Pull request merged**